snap-binning/src/test/java/org/esa/snap/binning/aggregators/AggregatorMeanObsTest.java

```diff
@@ -0,0 +1,316 @@
+        assertNotNull(aggregator);
+
+        assertEquals("variable_mean", aggregator.getTemporalFeatureNames()[0]);
+        assertEquals("variable_mean", aggregator.getOutputFeatureNames()[0]);
+    }
+
+    @Test
+    public void testDescriptor_createAggregator_withOutputNamed() {
+        final AggregatorMeanObs.Descriptor descriptor = new AggregatorMeanObs.Descriptor();
+
+        final Aggregator aggregator = descriptor.createAggregator(new MyVariableContext(), new AggregatorMeanObs.Config("variable", "Lena"));
+        assertNotNull(aggregator);
+
+        assertEquals("Lena_mean", aggregator.getTemporalFeatureNames()[0]);
+        assertEquals("Lena_mean", aggregator.getOutputFeatureNames()[0]);
+    }
+
+    @Test
+    public void testDescriptor_createConfig() {
+        final AggregatorMeanObs.Descriptor descriptor = new AggregatorMeanObs.Descriptor();
+
+        final AggregatorConfig aggregatorConfig = descriptor.createConfig();
+        assertNotNull(aggregatorConfig);
+
+        final AggregatorMeanObs.Config config = (AggregatorMeanObs.Config) aggregatorConfig;
+        assertNull(config.varName);
+        assertNull(config.targetName);
+        assertEquals("MEAN_OBS", config.getName());
+    }
+
+    @Test
+    public void testDescriptor_getSourceVarNames() {
+        final AggregatorMeanObs.Descriptor descriptor = new AggregatorMeanObs.Descriptor();
+
+        final String[] sourceVarNames = descriptor.getSourceVarNames(new AggregatorMeanObs.Config("source", "target"));
+        assertEquals(1, sourceVarNames.length);
+        assertEquals("source", sourceVarNames[0]);
+    }
+
+    @Test
+    public void testDescriptor_getTargetVarNames_standard() {
+        final AggregatorMeanObs.Descriptor descriptor = new AggregatorMeanObs.Descriptor();
+
+        final String[] sourceVarNames = descriptor.getTargetVarNames(new AggregatorMeanObs.Config("source", null));
+        assertEquals(3, sourceVarNames.length);
+        assertEquals("source_mean", sourceVarNames[0]);
+        assertEquals("source_sigma", sourceVarNames[1]);
+        assertEquals("source_counts", sourceVarNames[2]);
+    }
+
+    @Test
+    public void testDescriptor_getTargetVarNames_withOutputName() {
+        final AggregatorMeanObs.Descriptor descriptor = new AggregatorMeanObs.Descriptor();
+
+        final String[] sourceVarNames = descriptor.getTargetVarNames(new AggregatorMeanObs.Config("source", "Klaus"));
+        assertEquals(3, sourceVarNames.length);
+        assertEquals("Klaus_mean", sourceVarNames[0]);
+        assertEquals("Klaus_sigma", sourceVarNames[1]);
+        assertEquals("Klaus_counts", sourceVarNames[2]);
+    }
+
+    @Test
+    public void testDescriptor_getName() {
+        final AggregatorMeanObs.Descriptor descriptor = new AggregatorMeanObs.Descriptor();
+
+        assertEquals("MEAN_OBS", descriptor.getName());
+    }
+}
```

Comment:
The following test should succeed for this aggregator and fail for AVG. AVG would get 2.25 as mean, not 2.0.
   @Test
    public void testObs() {
        AggregatoMeanObs agg = new AggregatorMeanObs(new MyVariableContext("c"), "c", 0.0);

        VectorImpl svec1 = vec(NaN, NaN);
        VectorImpl svec2 = vec(NaN, NaN);
        VectorImpl tvec = vec(NaN, NaN, NaN);
        VectorImpl out = vec(NaN, NaN);

        agg.initSpatial(ctx, svec1);
        agg.aggregateSpatial(ctx, obsNT(1f), svec1);
        agg.aggregateSpatial(ctx, obsNT(2f), svec1);
        agg.completeSpatial(ctx, 2, svec1);

        agg.initSpatial(ctx, svec2);
        agg.aggregateSpatial(ctx, obsNT(3f), svec2);
        agg.completeSpatial(ctx, 1, svec2);

        agg.initTemporal(ctx, tvec);
        agg.aggregateTemporal(ctx, svec1, 2, tvec);
        agg.aggregateTemporal(ctx, svec2, 1, tvec);
        agg.completeTemporal(ctx, 2, tvec);
        agg.computeOutput(tvec, out);

        float mean = (1f + 2f + 3f) / 3;
        float sigma = (float) sqrt((1f*1f + 2f*2f + 3f*3f) / 3 - mean * mean);
        assertEquals(mean, out.get(0), 1e-5f);
        assertEquals(sigma, out.get(1), 1e-5f);
    }

Reply:
I can not see the difference to the tests already implemented - except for the numerical values. All this functionality is already tested exhaustively. What is the reason for this specific case?

Reply:
Which of your former tests would fail with AVG and succeed with the new aggregator? (Maybe, I overlooked it)

Reply:
testAggregateSpatial_threeMeasurement_withInvalid()

Reply:
Ah, I see. 
My test would have tested the complete scope of the aggregator (its "contract") without prescribing the implementation of the intermediate steps.